sympy/solvers/polysys.py

```diff
@@ -429,3 +439,428 @@ def solve_triangulated(polys, *gens, **args):
 
         solutions = _solutions
     return sorted((s for s, _ in solutions), key=default_sort_key)
+
+
+def factor_system(eqs, *gens, **kwargs):
+    """
+    Factorizes a system of polynomial equations into
+    irreducible subsystems.
+
+    Parameters
+    ==========
+
+    eqs : list
+        List of expressions to be factored.
+        Each expression is assumed to be equal to zero.
+
+    *gens : Symbol or sequence of Symbols, optional
+        Generator(s) of the polynomial ring.
+        If not provided, all free symbols will be used.
+
+    **kwargs : dict, optional
+        Same optional arguments taken by ``factor``
+
+    Returns
+    =======
+
+    tuple
+        A pair (systems, condition) where:
+        - systems is a list of lists of expressions,
+          where each sublist when solved gives one
+          component of the solution
+
+        - condition is a Boolean expression that must
+          be satisfied for the system to be solvable
```

Comment:
The `factor_system` docstring still needs an example to show when `cond` is not just true.